cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/io/cdap/cdap/datapipeline/draft/DraftService.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package io.cdap.cdap.datapipeline.draft;
+
+import com.google.common.base.Strings;
+import io.cdap.cdap.api.NamespaceSummary;
+import io.cdap.cdap.api.metrics.Metrics;
+import io.cdap.cdap.etl.common.Constants;
+import io.cdap.cdap.etl.proto.v2.ETLConfig;
+import io.cdap.cdap.spi.data.table.field.FieldType;
+import io.cdap.cdap.spi.data.transaction.TransactionRunner;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+
+/**
+ * Class to interact with the DraftStore
+ */
+public class DraftService {
+  private static final Logger LOG = LoggerFactory.getLogger(DraftService.class);
+  private final Metrics metrics;
+  private final DraftStore store;
+
+  public DraftService(TransactionRunner context, Metrics metrics) {
+    this.store = new DraftStore(context);
+    this.metrics = metrics;
+    if (metrics == null) {
+      LOG.warn("Metrics collector was not injected into DraftHandler");
+    }
+  }
+
+  /**
+   * Returns a sorted and filtered list of drafts for the given namespace and owner
+   *
+   * @param namespaceSummary the namespace to fetch the drafts from
+   * @param owner the id of the owner of the drafts
+   * @param includeConfig the returned Draft objects will not include the pipeline config if this is false
+   * @param sortRequest The sorting that should be applied to the results, pass null if no sorting is required. Note
+   *   that the sortField matches against the column names in the table spec found in {@link DraftStore}
+   * @param filter string used to do case-insensitive prefix matching on the draft name
+   * @return List of {@link Draft}
+   * @throws RuntimeException when an error occurs while fetching from the table
+   */
+  public List<Draft> listDrafts(NamespaceSummary namespaceSummary, String owner,
+                                boolean includeConfig, SortRequest sortRequest,
+                                @Nullable String filter) {
+    List<Draft> drafts = store.listDrafts(namespaceSummary, owner, sortRequest);
+    if (!includeConfig) {
```

Comment:
Why not have this passed to the store.listDrafts method and have the returned drafts already without the config?

Reply:
done